src/tile.rs

```diff
@@ -45,9 +45,59 @@ pub fn calc_tile_id(z: u8, x: u64, y: u64) -> u64 {
     base_id + tile_id
 }
 
+#[must_use]
+pub(crate) fn xyz(tile_id: u64) -> (u8, u64, u64) {
+    if tile_id == 0 {
+        return (0, 0, 0);
+    }
+
+    // Find the zoom level by comparing against pyramid sizes
+    let mut z = 0u8;
+    for (zoom, &pyramid_size) in PYRAMID_SIZE_BY_ZOOM.iter().enumerate() {
+        if tile_id < pyramid_size {
+            z = u8::try_from(zoom - 1).expect("malformed zoom level");
```

Comment:
can this panic be caused by the user input? If so, this should be an error, not a panic

Reply:
I added some comments to clarify -- let me know if there's a better way to clarify this...